chacra/async/recurring.py

```diff
@@ -84,21 +84,25 @@ def purge_repos(_now=None):
         project_repos = models.Repo.filter_by(project=p)
 
         # get the configuration for current project
-        ref_names = purge_rotation[project_name].keys()
         purge_project = purge_rotation.get(project_name)
-
-        for ref_name in ref_names:
-            purge_ref = purge_project[ref_name]
-            lifespan = now - datetime.timedelta(days=purge_ref.get('days', 14))
-            keep_minimum = purge_ref.get('keep_minimum', 0)
-            ref_repos = project_repos.filter_by(ref=ref_name)
-
-            # filter the query further, with the offset being the
-            # minimum to keep, and lifespan how old they should be
-            repos = ref_repos.filter(
-                    models.Repo.modified < lifespan).order_by(
-                        desc(models.Repo.modified)).offset(keep_minimum).all()
-            delete_repositories(repos, lifespan, keep_minimum)
+        # can be ref and/or flavor
+        for proj_filter, repo_attrs in purge_project.items():
+            for attr in repo_attrs:
+                purge_attr = repo_attrs[attr]
+                lifespan = now - datetime.timedelta(days=purge_attr.get('days', 14))
+                keep_minimum = purge_attr.get('keep_minimum', 0)
+                # add similar conditional to filter_by other repo attributes
+                if proj_filter == 'ref':
+                    attr_repos = project_repos.filter_by(ref=attr)
+                elif proj_filter == 'flavor':
```

Comment:
this `elif` means that if the first `if` is met (when `ref` is defined) this will never be executed. And I think you really want to allow filtering by both (this would need a test too, that filters by both)